app/code/Magento/Theme/Controller/Result/JsFooterPlugin.php

```diff
@@ -32,45 +32,52 @@ public function __construct(ScopeConfigInterface $scopeConfig)
     }
 
     /**
-     * Put all javascript to footer before sending the response.
+     * Moves all JavaScript tags to the end of body if this feature is enabled.
      *
      * @param Http $subject
      * @return void
      */
     public function beforeSendResponse(Http $subject)
     {
         $content = (string)$subject->getContent();
+        $bodyEndTag = '</body';
+        $isEndBodyTagFound = strpos($content, $bodyEndTag) !== false;
+        $shouldMoveJsToBottom = $this->scopeConfig->isSetFlag(self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM, ScopeInterface::SCOPE_STORE);
 
-        $bodyClose = '</body';
-
-        if (strpos($content, $bodyClose) !== false && $this->scopeConfig->isSetFlag(
-            self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
-            ScopeInterface::SCOPE_STORE
-        )) {
-            $scripts = '';
-            $scriptOpen = '<script';
-            $scriptClose = '</script>';
-            $scriptOpenPos = strpos($content, $scriptOpen);
+        if ($isEndBodyTagFound && $shouldMoveJsToBottom) {
+            if ($scripts = $this->extractScriptTags($content)) {
+                $content = str_replace($bodyEndTag, "$scripts\n$bodyEndTag", $content);
```

Comment:
Position of the closing body tag has already been determined above. There's not need to repeat the search as part of the replacement. Script tags can be inserted at a known position.

Recommended optimization:
```
$bodyEndTagPos = stripos($content, $bodyEndTag);
$isEndBodyTagFound = $bodyEndTagPos !== false;
// ...
$content = substr_replace($content, $scripts . "\n", $bodyEndTagPos, 0);
```

Reply:
@sshymko could we get this PR merged, and further improvements in separate one? :-)

Reply:
Feel free to ignore my comments. I'm not part of the Magento team :)

Reply:
We neither. We contribute to Magento just like you without being part of the structures

Reply:
@sshymko I applied your changes, the only issue I faced was the fact that previously fetched `bodyEndTagPos` changes due to `<script`s pulled out of DOM. So I'm going to slightly change the code to reflect the needs.